.circleci/config.yml

```diff
@@ -23,7 +23,61 @@ jobs:
             brew install shellcheck
             bundle check || bundle install --jobs=4 --retry=3 --path .bundle
       - save_cache:
-          key: v1-gems-{{ checksum "Gemfile" }}
+          key: fastlane-macos-gems-{{ checksum "Gemfile" }}
+          paths:
+            - .bundle
+
+      - run:
+          name: Check PR Metadata
+          command: bundle exec danger || echo "danger failed"
+
+      - run: bundle exec fastlane test
+
+      - store_test_results:
+          path: /Users/distiller/test-results
+
+      - run:
+          name: Post Test Results to GitHub
+          command: bundle exec danger || echo "danger failed"
+          when: always # Run this even when tests fail
+  
+  "Ubuntu":
+    environment:
+      CIRCLE_TEST_RESULTS: $HOME/test-results
+      LC_ALL: C.UTF-8
+      LANG: C.UTF-8
+      FASTLANE_ITUNES_TRANSPORTER_PATH: .bundle
+    docker:
+      - image: circleci/ruby:2.3
+    shell: /bin/bash --login -eo pipefail
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: fastlane-ubuntu-gems-{{ checksum "Gemfile" }}
+      - run:
+          name: Setup Build
+          command: |
+            mkdir -p $CIRCLE_TEST_RESULTS
+            echo "2.3" > .ruby-version
+            gem update --system
+            bundle check || bundle install --jobs=4 --retry=3 --path .bundle
+            sudo apt-get install shellcheck
+      - run:
+          name: Install additional dependency xar
+          command: |
+            cd /tmp
+            mkdir download
+            cd download
+            wget https://github.com/downloads/mackyle/xar/xar-1.6.1.tar.gz
+            tar -xzf xar-1.6.1.tar.gz
+            cd xar-1.6.1
+            ./autogen.sh --noconfigure
+            ./configure
+            make
+            sudo make install
+      - save_cache:
+          key: fastlane-ubuntu-gems-{{ checksum "Gemfile" }}
```

Comment:
does /Users/distiller work on Linux? (store_test_results section)

Reply:
afaik it doesn't even work on macOS - or I couldn't find out how it should work, what it should do etc. Do you have more insight?

Reply:
/Users/distiller will not work on linux

Reply:
Ok, looked into this a bit:

This step is currently failing on our normal builds:
https://circleci.com/gh/fastlane/fastlane/12636
```
Unable to save test results from /Users/distiller/test-results
Error stat /Users/distiller/test-results: no such file or directory
Found no path with test results, skipping
```

But test results should actually be written to somewhere: https://github.com/fastlane/fastlane/blob/3830c51f153875ec0b7350af359709e32616cfab/Rakefile#L18

I will create an issue for this first, when it is fixed the result can be incorporated here.

Reply:
This is now #11303 and PR #11304

Reply:
PR #11304 is ready for review (and merge). It fixes this step and will probably work unchanged on Ubuntu as well, so can be copied over.